### PR TITLE
[rai] drm check and fix

### PR DESCRIPTION
**Pull request by nixxo**
### Before submitting a *pull request* make sure you have:
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Read [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site)
- [x] Read [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) and adjusted the code to meet them
- [x] Covered the code with tests (note that PRs without tests will be REJECTED)
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [x] Improvement
---

### Description of your *pull request* and other information

- Add DRM check support that raises an error instead of extractor outputting an encrypted dash manifest
- Fix for #12652

### CODE REVIEW

**Comment by dstftw** [comment]
Read coding conventions.

File: youtube_dl/extractor/rai.py [hunk 1]
```diff
@@ -410,6 +439,11 @@ def _real_extract(self, url):
                 ''' % self._UUID_RE,
                 webpage, 'content item id', default=None, group='id')
 
+        if not content_item_id:
+            content_item_id = self._search_regex(
+                r'/ContentItem-(?P<id>%s)\.html\?iframe' % self._UUID_RE,
+                webpage, 'content item id', default=None, group='id')
+
```
Comment (dstftw) [inline youtube_dl/extractor/rai.py:446]: Do not mix unrelated changes in single PR.

File: youtube_dl/extractor/rai.py [hunk 2]
```diff
@@ -145,14 +145,30 @@ class RaiPlayIE(RaiBaseIE):
...
     }]
 
+    def _check_drm(self, obj):
+        if try_get(obj, lambda x: x['rights_management']['rights']['drm'], dict):
+            return True
+
+        if obj.get('program_info'):
+            return self._check_drm(obj['program_info'])
+
+        return False
```
Comment (dstftw) [inline youtube_dl/extractor/rai.py:161]: Inline.

**Review by dstftw — CHANGES_REQUESTED** [review]

File: youtube_dl/extractor/rai.py (hunk 1 above)
Comment (nixxo) [reply youtube_dl/extractor/rai.py]: here: #27673

File: youtube_dl/extractor/rai.py (hunk 2 above)
Comment (nixxo) [reply youtube_dl/extractor/rai.py]: done in 614c48f3f497cb9537319057fadb28623d5f9d87

**Comment by dirkf** [comment]
Is this still relevant?

**Comment by nixxo** [comment]
no, it can be closed.

**Comment by dirkf** [comment]
Thx

**Comment by nixxo** [comment]
@dirkf sorry, my mistake.
I thought I was looking at yt-dlp PRs and not the original youtube-dl. This PR should be re-opened and merged.

Sorry again.

**Comment by dirkf** [comment]
The linked issue was marked as fixed in https://github.com/ytdl-org/youtube-dl/commit/fcd90d258305cdafa5bd23c50443229205fcb9ed (your #27673) but Sergey didn't merge this change.

I distilled it to this:
```py
--- a/youtube_dl/extractor/rai.py
+++ b/youtube_dl/extractor/rai.py
@@ -158,6 +158,10 @@ class RaiPlayIE(RaiBaseIE):
         # subtitles at 'subtitlesArray' key (see #27698)
         'url': 'https://www.raiplay.it/video/2020/12/Report---04-01-2021-2e90f1de-8eee-4de4-ac0e-78d21db5b600.html',
         'only_matching': True,
+    }, {
+        # DRM protected
+        'url': 'https://www.raiplay.it/video/2020/09/Lo-straordinario-mondo-di-Zoey-S1E1-Lo-straordinario-potere-di-Zoey-ed493918-1d32-44b7-8454-862e473d00ff.html',
+        'only_matching': True,
     }]
 
     def _real_extract(self, url):
@@ -166,6 +170,13 @@ class RaiPlayIE(RaiBaseIE):
         media = self._download_json(
             base + '.json', video_id, 'Downloading video JSON')
 
+        if try_get(
+                media,
+                (lambda x: x['rights_management']['rights']['drm'],
+                 lambda x: x['program_info']['rights_management']['rights']['drm']),
+                dict):
+             raise ExtractorError('This video is DRM protected.', expected=True)
+
         title = media['name']
 
         video = media['video']
```

**Comment by nixxo** [comment]
dstftw asked me to split the PR in 2 because it fixed two different problems (see here: https://github.com/ytdl-org/youtube-dl/pull/27657#discussion_r551657747) and so I did (#27657 #27673) but then he only merged one leaving this one un-merged.
